Add documentation for southbound providers (#170)
eclipse/org.eclipse.sensinact.gateway: prototype/docs/southbound/device-factory/csv.md

# CSV parser for the device factory

Eclipse sensiNact provides a CSV parser for its device factory.
It uses the parser ID `csv`.

## Bundles

The CSV parser bundle is: `org.eclipse.sensinact.gateway.southbound.device-factory:parser-csv:0.0.2`.

It requires the following bundles to be resolved:
* the device factory core
* the Apache Commons CSV parsing library `org.apache.commons:commons-csv:1.9.0`.

## Parser paths

> If the CSV payload accepts integer column index as path, with the first column having index 0.

Reviewer: ```suggestion
The CSV data can be mapped using the integer column index as the path, with the first column having index 0.
```

> If the payload has a header, the column name can also be used as path. If multiple columns have the same name, the leftmost one will be used.

Reviewer: ```suggestion
If the CSV data has a header row then the column name can also be used as the path. If multiple columns have the same name, the first one encountered will be used.
```
